superbench/executor/executor.py

```diff
@@ -106,17 +110,47 @@ def __exec_benchmark(self, context, log_suffix):
                     logger.info('Executor succeeded in %s.', log_suffix)
                 else:
                     logger.error('Executor failed in %s.', log_suffix)
+                return benchmark.result
             else:
                 logger.error('Executor failed in %s, invalid context.', log_suffix)
-        except Exception:
+        except Exception as e:
+            logger.error(e)
             logger.error('Executor failed in %s.', log_suffix)
+        return None
+
+    def __get_benchmark_dir(self, benchmark_name, create=False):
+        """Get output directory for benchmark.
+
+        Args:
+            benchmark_name (str): Benchmark name.
+            create (bool): Create the directory or not.
+
+        Return:
+            str: Benchmark output directory.
+        """
+        benchmark_output_dir = Path(self._output_dir, 'benchmarks', benchmark_name)
+        if create:
+            benchmark_output_dir.mkdir(mode=0o755, parents=True, exist_ok=True)
+        return str(benchmark_output_dir)
+
+    def __write_benchmark_results(self, benchmark_name, benchmark_results):
+        """Write benchmark results.
+
+        Args:
+            benchmark_name (str): Benchmark name.
+            benchmark_results (dict): Benchmark results.
+        """
+        with Path(self.__get_benchmark_dir(benchmark_name), 'results.json').open(mode='w') as f:
```

Comment:
Is it a separate json file?

Reply:
every benchmark will have a results.json in its directory
we may need a merged json after runner collected all json on all nodes

Reply:
I think one json should be good.

Reply:
Besides, based on our previous bad experience, if we do the merge offline. We don't need to do this in this version. We should do it in the runner.

Reply:
will add in runner in another pr